tests/nosetests/pyanaconda_tests/module_source_harddrive_test.py

```diff
@@ -0,0 +1,277 @@
+           return_value="image_file.iso")
+    def success_find_iso_test(self, find_first_iso_image_mock, mount_mock, device_matches_mock):
+        """Hard drive source setup iso found."""
+        task = SetUpHardDriveSourceTask(mount_location, "partition", "direc/tory")
+        result = task.run()
+
+        device_matches_mock.assert_called_once_with("partition")
+        find_first_iso_image_mock.assert_called_once_with(
+            mount_location + "_device/direc/tory"
+        )
+        mount_mock.assert_has_calls([
+            call("/dev/part", mount_location + "_device", "auto", ""),
+            call(mount_location + "_device/direc/tory/image_file.iso", mount_location + "_iso",
+                 fstype='iso9660', options="ro")
+        ])
+        self.assertEqual(result, mount_location + "_iso")
+
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.device_matches",
+           return_value=["part"])
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.mount")
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.find_first_iso_image",
+           return_value=None)
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.verify_valid_installtree",
+           return_value=True)
+    def success_find_dir_test(self, verify_valid_installtree_mock, find_first_iso_image_mock,
+                              mount_mock, device_matches_mock):
+        """Hard drive source setup dir found."""
+        task = SetUpHardDriveSourceTask(mount_location, "partition", "direc/tory")
+        result = task.run()
+        device_matches_mock.assert_called_once_with("partition")
+        find_first_iso_image_mock.assert_called_once_with(
+            mount_location + "_device/direc/tory"
+        )
+        mount_mock.assert_called_once_with("/dev/part", mount_location + "_device", "auto", "")
+        verify_valid_installtree_mock.assert_called_once_with(
+            mount_location + "_device/direc/tory"
+        )
+        self.assertEqual(result, mount_location + "_device/direc/tory")
+
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.device_matches",
+           return_value=["part"])
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.mount")
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.find_first_iso_image",
+           return_value=None)
+    @patch("pyanaconda.modules.payloads.source.harddrive.initialization.verify_valid_installtree",
+           return_value=False)
+    def failure_to_find_test(self, verify_valid_installtree_mock, find_first_iso_image_mock,
+                             mount_mock, device_matches_mock):
+        """Hard drive source setup failure to find anything."""
+        task = SetUpHardDriveSourceTask(mount_location, "partition", "direc/tory")
+        with self.assertRaises(SourceSetupError) as cm:
```

Comment:
```suggestion
        with self.assertRaises(SourceSetupError):
```

Do not take exception when it is not used anywhere.

Reply:
It's 10 lines below.

Reply:
Ohh in that case I've missed that. My fault.